lib/api/controllers/securityController.js

```diff
@@ -845,21 +845,19 @@ class SecurityController extends NativeController {
       throw kerror.get('api', 'assert', 'invalid_type', 'body.content.profileIds', 'array');
     }
 
-    try {
-      return await this._changeUser(request, id, content, userId, profileIds);
-    }
-    catch (error) {
-      if (error.id && error.id === 'services.storage.not_found') {
-        const creatingContent = {
-          ...defaultValues,
-          ...content, // Order important, content erase default duplicates
-        };
-
-        return this._persistUser(request, profileIds, creatingContent);
-      }
-
-      throw error;
-    }
+    return this._changeUser(request, id, content, userId, profileIds)
```

Comment:
Why changing to promise chain? `async/await` should be preferred

Reply:
Since the function would send a Promise, I thought that `then` would be preferred as it does not block waiting for the result (or error) of the promise

Reply:
It will not change anything because the result of the `upsertUser` will not be available until the result of the `_changeUser` method is available (=resolved from the promise).

So you can await the promise resolution in this method with `await` or return a promise who will be awaited later by another method. It does not change the way any subsequent caller will be blocked when having to wait for the promise to be resolved.